[CI] BulkStoredSourceChallengeRestIT testEsqlTermsAggregationByMethod failing

**Build Scans:**
- [elasticsearch-periodic #10725 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/lgoav7675l4ka)
- [elasticsearch-pull-request #93344 / part-2](https://gradle-enterprise.elastic.co/s/nnbsrq4pl4lei)
- [elasticsearch-pull-request #93191 / part-2](https://gradle-enterprise.elastic.co/s/ywxx4fzqtvhxc)
- [elasticsearch-periodic-platform-support #10679 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/b6bam25d2tluc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:logsdb:javaRestTest" --tests "org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT.testEsqlTermsAggregationByMethod" -Dtests.seed=5DAD5247074B05DA -Dtests.locale=kok-Latn -Dtests.timezone=America/Bahia -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEsqlTermsAggregationByMethod'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: errors in baseline bulk response:
 {took=1000, items=[{create={result=created, _shards={total=2, failed=0, successful=1}, _seq_no=0, _index=.ds-standard-apache-baseline-2025.09.20-000001, forced_refresh=true, _id=0, _version=1, _primary_term=1, status=201}}, {create={result=created, _shards={total=2, failed=0, successful=1}, _seq_no=0, _index=.ds-standard-apache-baseline-2025.09.20-000001, forced_refresh=true, _id=1, _version=1, _primary_term=1, status=201}}, {create={result=created, _shards={total=2, failed=0, successful=1}, _seq_no=1, _index=.ds-standard-apache-baseline-2025.09.20-000001, forced_refresh=true, _id=2, _version=1, _primary_term=1, status=201}}, {create={result=created, _shards={total=2, failed=0, successful=1}, _seq_no=2, _index=.ds-standard-apache-baseline-2025.09.20-000001, forced_refresh=true, _id=3, _version=1, _primary_term=1, status=201}}, {create={result=created, _shards={total=2, failed=0, successful=1}, _seq_no=0, _index=.ds-standard-a
[truncated]
```

**Issue Reasons:**
- [main] 4 failures in test testEsqlTermsAggregationByMethod (0.8% fail rate in 491 executions)
- [main] 2 failures in step part-2 (0.8% fail rate in 245 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.8% fail rate in 243 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java
+++ b/libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java
@@ -16,7 +16,6 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.JsonToken;
 import com.fasterxml.jackson.core.exc.InputCoercionException;
 import com.fasterxml.jackson.core.exc.StreamConstraintsException;
-import com.fasterxml.jackson.core.filter.FilteringParserDelegate;
 import com.fasterxml.jackson.core.io.JsonEOFException;
 
 import org.elasticsearch.core.IOUtils;
@@ -27,7 +26,6 @@ import org.elasticsearch.xcontent.XContentParseException;
 import org.elasticsearch.xcontent.XContentParserConfiguration;
 import org.elasticsearch.xcontent.XContentString;
 import org.elasticsearch.xcontent.XContentType;
-import org.elasticsearch.xcontent.provider.OptimizedTextCapable;
 import org.elasticsearch.xcontent.provider.XContentParserConfigurationImpl;
 import org.elasticsearch.xcontent.support.AbstractXContentParser;
 

--- a/libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java
+++ b/libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java
@@ -145,19 +145,7 @@ public class JsonXContentParser extends AbstractXContentParser {
 
     @Override
     public XContentString optimizedText() throws IOException {
-        if (currentToken().isValue() == false) {
-            throwOnNoText();
-        }
-        var parser = this.parser;
-        if (parser instanceof FilteringParserDelegate delegate) {
-            parser = delegate.delegate();
-        }
-        if (parser instanceof OptimizedTextCapable optimizedTextCapableParser) {
-            var bytesRef = optimizedTextCapableParser.getValueAsText();
-            if (bytesRef != null) {
-                return bytesRef;
-            }
-        }
+        // TODO: enable utf-8 parsing optimization once verified it is completely safe
         return new Text(text());
     }
 

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -618,48 +618,12 @@ tests:
 - class: org.elasticsearch.search.vectors.CachingEnableFilterQueryTests
   method: testBooleanQuery
   issue: https://github.com/elastic/elasticsearch/issues/135124
-- class: org.elasticsearch.xpack.logsdb.qa.LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT
-  method: testEsqlTermsAggregationByMethod
-  issue: https://github.com/elastic/elasticsearch/issues/135125
-- class: org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT
-  method: testEsqlTermsAggregationByMethod
-  issue: https://github.com/elastic/elasticsearch/issues/135126
-- class: org.elasticsearch.xpack.logsdb.qa.StandardVersusStandardReindexedIntoLogsDbChallengeRestIT
-  method: testEsqlTermsAggregationByMethod
-  issue: https://github.com/elastic/elasticsearch/issues/135127
-- class: org.elasticsearch.xpack.logsdb.qa.BulkChallengeRestIT
-  method: testEsqlTermsAggregationByMethod
-  issue: https://github.com/elastic/elasticsearch/issues/135128
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=transform/transforms_start_stop/Test stop transform with force and wait_for_checkpoint true}
   issue: https://github.com/elastic/elasticsearch/issues/135135
 - class: org.elasticsearch.compute.aggregation.SampleIntAggregatorFunctionTests
   method: testSimpleWithCranky
   issue: https://github.com/elastic/elasticsearch/issues/135137
-- class: org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT
-  method: testTermsAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135140
-- class: org.elasticsearch.xpack.logsdb.qa.LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT
-  method: testTermsAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135141
-- class: org.elasticsearch.xpack.logsdb.qa.BulkChallengeRestIT
-  method: testTermsAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135142
-- class: org.elasticsearch.xpack.logsdb.qa.StandardVersusStandardReindexedIntoLogsDbChallengeRestIT
-  method: testTermsAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135143
-- class: org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT
-  method: testMatchAllQuery
-  issue: https://github.com/elastic/elasticsearch/issues/135144
-- class: org.elasticsearch.xpack.logsdb.qa.BulkChallengeRestIT
-  method: testMatchAllQuery
-  issue: https://github.com/elastic/elasticsearch/issues/135145
-- class: org.elasticsearch.xpack.logsdb.qa.LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT
-  method: testMatchAllQuery
-  issue: https://github.com/elastic/elasticsearch/issues/135146
-- class: org.elasticsearch.xpack.logsdb.qa.StandardVersusStandardReindexedIntoLogsDbChallengeRestIT
-  method: testMatchAllQuery
-  issue: https://github.com/elastic/elasticsearch/issues/135147
 - class: org.elasticsearch.lucene.RollingUpgradeSearchableSnapshotIndexCompatibilityIT
   method: testSearchableSnapshotUpgrade {p0=[9.2.0, 9.2.0, 9.2.0]}
   issue: https://github.com/elastic/elasticsearch/issues/135150
@@ -669,18 +633,6 @@ tests:
 - class: org.elasticsearch.discovery.ClusterDisruptionIT
   method: testAckedIndexing
   issue: https://github.com/elastic/elasticsearch/issues/117024
-- class: org.elasticsearch.xpack.logsdb.qa.BulkChallengeRestIT
-  method: testHistogramAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135153
-- class: org.elasticsearch.xpack.logsdb.qa.LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT
-  method: testHistogramAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135154
-- class: org.elasticsearch.xpack.logsdb.qa.StandardVersusStandardReindexedIntoLogsDbChallengeRestIT
-  method: testHistogramAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135155
-- class: org.elasticsearch.xpack.logsdb.qa.BulkStoredSourceChallengeRestIT
-  method: testHistogramAggregation
-  issue: https://github.com/elastic/elasticsearch/issues/135156
 - class: org.elasticsearch.compute.aggregation.SampleBytesRefAggregatorFunctionTests
   method: testSimpleWithCranky
   issue: https://github.com/elastic/elasticsearch/issues/135157

--- a/server/src/test/java/org/elasticsearch/common/xcontent/json/JsonXContentTests.java
+++ b/server/src/test/java/org/elasticsearch/common/xcontent/json/JsonXContentTests.java
@@ -59,7 +59,8 @@ public class JsonXContentTests extends BaseXContentTestCase {
             assertSame(XContentParser.Token.FIELD_NAME, parser.nextToken());
             assertTrue(parser.nextToken().isValue());
             Text text = (Text) parser.optimizedText();
-            assertTrue(text.hasBytes());
+            // TODO: uncomment after utf8 optimized parsing has been enabled again:
+//            assertTrue(text.hasBytes());
             assertThat(text.string(), equalTo("foo"));
         }
     }

--- a/server/src/test/java/org/elasticsearch/common/xcontent/json/JsonXContentTests.java
+++ b/server/src/test/java/org/elasticsearch/common/xcontent/json/JsonXContentTests.java
@@ -60,7 +60,7 @@ public class JsonXContentTests extends BaseXContentTestCase {
             assertTrue(parser.nextToken().isValue());
             Text text = (Text) parser.optimizedText();
             // TODO: uncomment after utf8 optimized parsing has been enabled again:
-//            assertTrue(text.hasBytes());
+            // assertTrue(text.hasBytes());
             assertThat(text.string(), equalTo("foo"));
         }
     }

